search-app/src/cmr/search/site/data.clj

```diff
@@ -28,16 +28,39 @@
   (case tag
     "gov.nasa.eosdis" "EOSDIS"))
 
+(defn collection-data
+  "Get the collection data associated with a provider and tag."
+  [context tag provider-id]
+  (->> {:tag-key tag
+        :provider provider-id
+        :result-format {:format :umm-json-results}}
+       (query-svc/make-concepts-query context :collection)
+       (query-exec/execute-query context)
+       :items))
+
 (defn provider-data
   "Create a provider data structure suitable for template iteration to
   generate links.
 
   Note that the given tag will be used to filter provider collections data
   that is used on the destination page."
-  [tag provider-data]
-  {:id (:provider-id provider-data)
-   :name (:provider-id provider-data)
-   :tag tag})
+  [context tag data]
+  (let [provider-id (:provider-id data)
+        provider-name provider-id
```

Comment:
provider has a `short-name`. If we are putting a name out there, it probably should be the `short-name`.  I am ok with the html page just use the provider-id, then we can just drop `provider-name` from the data structure.